### PR TITLE
cephfs-journal-tool: make --rank argument mandatory

### PR DESCRIPTION
**Pull request by vshankar**

### CODE REVIEW

File: src/tools/cephfs/JournalTool.cc [hunk 1]
```diff
@@ -96,9 +94,7 @@ int JournalTool::main(std::vector<const char*> &argv)
 
   std::string rank_str;
   if (!ceph_argparse_witharg(argv, arg, &rank_str, "--rank", (char*)NULL)) {
-    // Default: act on rank 0.  Will give the user an error if they
-    // try invoking this way when they have more than one filesystem.
-    rank_str = "0";
+    return -EINVAL;
```
Comment (batrick) [inline src/tools/cephfs/JournalTool.cc]: Some kind of error message woudl be appropriate?

**Comment by vshankar** [comment]
@batrick updated

Also, wanted to check if we should disallow actions such as `journal import` for "all" ranks from a single import file (which probably was generated from `journal export`)?

**Comment by batrick** [comment]
> Also, wanted to check if we should disallow actions such as journal import for "all" ranks from a single import file (which probably was generated from journal export)?

Yes that makes sense. Thanks for catching that. Please also audit any other commands that would have similar issues.

**Comment by vshankar** [comment]
@batrick updated -- `journal import` looks like the only operation that needs to be blocked for operation on all ranks.

**Comment by vshankar** [comment]
retest this please

**Comment by batrick** [comment]
Delaying QA of this due to conflicts with #23467.

**Comment by vshankar** [comment]
rebased and updated

**Comment by ukernel** [comment]
need to update qa test cases

**Comment by vshankar** [comment]
fixed and updated -- tested with vstart_runner

**Comment by vshankar** [comment]
fixed and updated

initial teuthology run is here : http://pulpito.ceph.com/vshankar-2018-09-19_15:06:02-fs-wip-cephfs-tools-rank-ops-distro-basic-mira/

failures do not seem to be due to these changes. one failure that seemed problematic was this: http://pulpito.ceph.com/vshankar-2018-09-19_15:06:02-fs-wip-cephfs-tools-rank-ops-distro-basic-mira/3043417/ which is a failed asset as per below:

```
build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos7/DIST/centos7/MACHINE_SIZE/huge/release/14.0.0-3394-gab60dcf/rpm/el7/BUILD/ceph-14.0.0-3394-gab60dcf/src/osdc/Journaler.cc: 420: FAILED ceph_assert(!r)
2018-09-20T10:53:23.175 INFO:tasks.ceph.mds.a-s.mira114.stderr: ceph version 14.0.0-3394-gab60dcf (ab60dcf2e503deb2846c9a5855d42c8d4d834272) nautilus (dev)
2018-09-20T10:53:23.175 INFO:tasks.ceph.mds.a-s.mira114.stderr: 1: (ceph::__ceph_assert_fail(char const*, char const*, int, char const*)+0x153) [0x7fe687295f0f]
2018-09-20T10:53:23.175 INFO:tasks.ceph.mds.a-s.mira114.stderr: 2: (ceph::__ceph_assertf_fail(char const*, char const*, int, char const*, char const*, ...)+0) [0x7fe6872960dd]
2018-09-20T10:53:23.175 INFO:tasks.ceph.mds.a-s.mira114.stderr: 3: (()+0x3cd7ba) [0x5617736317ba]
2018-09-20T10:53:23.175 INFO:tasks.ceph.mds.a-s.mira114.stderr: 4: (Context::complete(int)+0x9) [0x5617733745d9]
2018-09-20T10:53:23.175 INFO:tasks.ceph.mds.a-s.mira114.stderr: 5: (Journaler::_finish_reread_head(int, ceph::buffer::list&, Context*)+0x2a6) [0x56177362f206]
2018-09-20T10:53:23.175 INFO:tasks.ceph.mds.a-s.mira114.stderr: 6: (Context::complete(int)+0x9) [0x5617733745d9]
2018-09-20T10:53:23.175 INFO:tasks.ceph.mds.a-s.mira114.stderr: 7: (Finisher::finisher_thread_entry()+0x11e) [0x7fe6872e0dfe]
2018-09-20T10:53:23.175 INFO:tasks.ceph.mds.a-s.mira114.stderr: 8: (()+0x7dd5) [0x7fe684f27dd5]
2018-09-20T10:53:23.176 INFO:tasks.ceph.mds.a-s.mira114.stderr: 9: (clone()+0x6d) [0x7fe683bdab3d]
2018-09-20T10:53:23.176 INFO:tasks.ceph.mds.a-s.mira114.stderr:*** Caught signal (Aborted) **
2018-09-20T10:53:23.176 INFO:tasks.ceph.mds.a-s.mira114.stderr: in thread 7fe676be0700 thread_name:fn_anonymous
```

... rerunning the above test passes: http://pulpito.ceph.com/vshankar-2018-09-21_04:27:30-fs-wip-cephfs-tools-rank-ops-distro-basic-mira/3049061/

**Pull request merged**